plugins/modules/content_view_filter_rule.py

```diff
@@ -0,0 +1,364 @@
+    content_view: "Standard Operating Environment"
+    content_view_filter: "modulemd filter"
+    name: "389-directory-server"
+    stream: "next"
+    version: "820220325123957"
+    context: "9edba152"
+    state: present
+'''
+
+RETURN = '''
+entity:
+  description: Final state of the affected entities grouped by their type.
+  returned: success
+  type: dict
+  contains:
+    content_view_filters:
+      description: List of content view filters.
+      type: list
+      elements: dict
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import KatelloMixin, ForemanStatelessEntityAnsibleModule
+
+content_filter_rule_erratum_spec = {
+    'id': {},
+    'date_type': {},
+    'end_date': {},
+    'start_date': {},
+    'types': {'type': 'list'},
+}
+
+content_filter_rule_erratum_id_spec = {
+    'id': {},
+    'errata_id': {},
+}
+
+content_filter_rule_rpm_spec = {
+    'id': {},
+    'rule_name': {'flat_name': 'name'},
+    'max_version': {},
+    'min_version': {},
+    'version': {},
+    'architecture': {},
+}
+
+content_filter_rule_modulemd_spec = {
+    'id': {},
+    'module_stream_ids': {'type': 'list'},
+}
+
+content_filter_rule_package_group_spec = {
+    'id': {},
+    'rule_name': {'flat_name': 'name'},
+    'uuid': {},
+}
+
+content_filter_rule_docker_spec = {
+    'id': {},
+    'rule_name': {'flat_name': 'name'},
+}
+
+
+class KatelloContentViewFilterRuleModule(KatelloMixin, ForemanStatelessEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = KatelloContentViewFilterRuleModule(
+        foreman_spec=dict(
+            content_view=dict(type='entity', scope=['organization'], required=True),
+            content_view_filter=dict(type='entity', scope=['content_view'], required=True),
+            name=dict(aliases=['rule_name', 'module_name', 'package_name', 'package_group', 'tag']),
+            state=dict(default='present', choices=['present', 'absent']),
+            errata_id=dict(),
+            types=dict(default=["bugfix", "enhancement", "security"], type='list', elements='str'),
+            date_type=dict(default='updated', choices=['issued', 'updated']),
+            start_date=dict(),
+            end_date=dict(),
+            architecture=dict(aliases=['arch']),
+            version=dict(),
+            max_version=dict(),
+            min_version=dict(),
+            stream=dict(),
+            context=dict(),
+        ),
+        entity_opts=dict(scope=['content_view_filter']),
+    )
+
+    # Do we want to make sure it exists or doesn't exist
+    rule_state = module.foreman_params.pop('state')
+
+    with module.api_connection():
+
+        # A filter always exists before we create a rule
+        # Get a reference to the content filter that owns the rule we want to manage
+        cv_scope = module.scope_for('content_view')
+        cvf_scope = module.scope_for('content_view_filter')
+        cvf = module.lookup_entity('content_view_filter', params=cv_scope)
```

Comment:
`params=cv_scope` should not be needed here - `lookup_entity` should be able to figure that out itself based on your `foreman_spec` above

Reply:
fixed